src/easyscience/variable/parameter.py

```
@@ -550,6 +595,68 @@ def _process_dependency_unique_names(self, dependency_expression: str):
                 raise ValueError(f'The object with unique_name {stripped_name} is not a Parameter or DescriptorNumber. Please check your dependency expression.') # noqa: E501
         self._clean_dependency_string = clean_dependency_string
 
+    def _convert_to_dict(self, d: dict, encoder, skip=None, **kwargs) -> dict:
+        """Custom serialization to handle parameter dependencies."""
+        if skip is None:
+            skip = []
+
+        # Add dependency information for dependent parameters
+        if not self._independent:
+            # Save the dependency expression
+            d['_dependency_string'] = self._dependency_string
+
+            # Convert dependency_map to use dependency_ids (preferred) and unique_names (fallback)
+            d['_dependency_map_dependency_ids'] = {}
+            d['_dependency_map_unique_names'] = {}
+
+            for key, dep_obj in self._dependency_map.items():
+                # Store dependency_id if available (more reliable)
+                if hasattr(dep_obj, '__dependency_id'):
+                    d['_dependency_map_dependency_ids'][key] = dep_obj.__dependency_id
+                # Also store unique_name as fallback
+                if hasattr(dep_obj, 'unique_name'):
+                    d['_dependency_map_unique_names'][key] = dep_obj.unique_name
+                else:
+                    # Last resort fallback for objects without unique_name
+                    d['_dependency_map_unique_names'][key] = str(dep_obj)
```

Would it perhaps be better to raise an error if the object doesn't have a unique name? A far as I know, every object should have a unique name, so if it doesn't, something has gone very wrong.

---

Yes, lack of unique_name should make the loader fail earlier already (hopefully). If there's no uniqe_name at this point, we should definitely throw.